lib/components/textEditor/richText.js

```diff
@@ -88,9 +90,26 @@ class RichText extends Component {
   focus = () => this.editor.focus();
 
   onChange = (editorState) => {
+    const {
+      maxContentSize: { bytes: maxBytes, warning },
+      getEditorData,
+    } = this.props;
+
     this.setState({ editorState }, () => {
-      // eslint-disable-next-line react/destructuring-assignment
-      this.props.getEditorData(stateToHTML(this.state.editorState.getCurrentContent()));
+      // eslint-disable-next-line no-shadow
+      const { warningModalWasShown, editorState } = this.state;
+      getEditorData(stateToHTML(editorState.getCurrentContent()));
```

Comment:
Nit: Unrelated to your PR, but your PR make it more apparent that this looks strange. It is hard to understand what this does on its own without knowing what uses this component. A comment could help, but again, not really related to you PR.